debezium-connector-mongodb/src/main/java/io/debezium/connector/mongodb/connection/client/MongoDbClientFactory.java

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright Debezium Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.debezium.connector.mongodb.connection.client;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.security.KeyManagementException;
+import java.security.KeyStore;
+import java.security.KeyStoreException;
+import java.security.NoSuchAlgorithmException;
+import java.security.UnrecoverableKeyException;
+import java.security.cert.CertificateException;
+
+import javax.net.ssl.KeyManager;
+import javax.net.ssl.KeyManagerFactory;
+import javax.net.ssl.SSLContext;
+import javax.net.ssl.TrustManager;
+import javax.net.ssl.TrustManagerFactory;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.mongodb.MongoClientSettings;
+import com.mongodb.client.MongoClient;
+import com.mongodb.client.MongoClients;
+
+import io.debezium.DebeziumException;
+import io.debezium.connector.mongodb.MongoDbConnectorConfig;
+
+public interface MongoDbClientFactory {
+
+    Logger LOGGER = LoggerFactory.getLogger(MongoDbClientFactory.class);
+
+    /**
+     * Creates {@link MongoClientSettings} used to obtain {@link MongoClient} instances
+     *
+     * @return client settings
+     */
+    MongoClientSettings getMongoClientSettings();
+
+    /**
+     * Creates native {@link MongoClient} instance
+     *
+     * @return mongo client
+     */
+    default MongoClient getMongoClient() {
+        var clientSettings = getMongoClientSettings();
+        return MongoClients.create(clientSettings);
+    }
+
+    /**
+     * Creates keystore
+     *
+     * @param path keyfile path
+     * @param password keyfile password
+     * @return keystore with loaded keys
+     */
+    static KeyStore loadKeyStore(Path path, char[] password) {
+        try (var keys = Files.newInputStream(path)) {
+            var ks = KeyStore.getInstance("PKCS12");
```

Comment:
@jpechane @Naros  this could actually also be configurable. So user should decide whether to use `jks` or `pkcs12` or whatever else.

Reply:
@jcechace ok, let's make it configurable. I think we'll take MongoDB as test bed from SSL config PoV that will be generalized and re-used accross all connectors.

Reply:
Done